tabu/src/tabu_search.cc

```diff
@@ -22,20 +22,14 @@
 using std::vector;
 using std::size_t;
 
-TabuSearch::TabuSearch(vector<vector<double>> Q, const vector<int> initSol, int tenure, long int timeout) : bqp(Q)
-{
-    size_t nvars = Q.size();
-    if (initSol.size() != nvars)
-        throw Exception("length of init_solution doesn't match the size of Q");
-
-    if (tenure < 0 || tenure > (nvars - 1))
-        throw Exception("tenure must be in the range [0, num_vars - 1]");
-    else if (tenure > 0) {
-        tabooTenure = tenure;
-    }
-    else {
-        tabooTenure = (20 < (int)(bqp.nVars / 4.0))? 20 : (int)(bqp.nVars / 4.0);
-    }
```

Comment:
Why remove these checks?

Reply:
All these checks are already done in sampler.py, so I figured that these were unneeded (Exceptions will never be thrown).

Reply:
But what happens if someone uses C++ code directly (or via cython), and doesn't validate parameters?

Reply:
Ah that's fair, I didn’t think we needed to take that into account. Will add these back!